main_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"flag"
+	"github.com/Shopify/sarama"
+	"github.com/Shopify/sarama/mocks"
+	"github.com/companieshouse/chs.go/kafka/producer"
+	. "github.com/smartystreets/goconvey/convey"
+	"testing"
+)
+
+func TestUnitProcessMessages(t *testing.T) {
+	offsetArraySingle := []int64{1}
+
+	Convey("test successful - message published", t, func() {
+		consumerMock := mocks.NewConsumer(t, nil)
+		consumerGroup := consumerMock.ExpectConsumePartition("", 0, 1)
+		consumerGroup.YieldMessage(&sarama.ConsumerMessage{})
+
+		producerMock := mocks.NewSyncProducer(t, nil)
+		producerMock.ExpectSendMessageAndSucceed()
```

Comment:
This test currently doesn't have any assertions, we should use `ExpectSendMessageWithCheckerFunctionAndSucceed` so that we can ensure the mock producer gets called as expected. For an example of using this you can check the sarama library (mocks folder) https://github.com/Shopify/sarama